sklearn/linear_model/_sgd_fast.pyx

```diff
@@ -514,11 +513,11 @@ def _plain_sgd(cnp.ndarray[double, ndim=1, mode='c'] weights,
     cdef unsigned char [:] validation_mask_view = validation_mask
 
     # q vector is only used for L1 regularization
-    cdef cnp.ndarray[double, ndim = 1, mode = "c"] q = None
+    cdef double[::1] q = None
     cdef double * q_data_ptr = NULL
     if penalty_type == L1 or penalty_type == ELASTICNET:
         q = np.zeros((n_features,), dtype=np.float64, order="c")
-        q_data_ptr = <double * > q.data
+        q_data_ptr = &q[0]
     cdef double u = 0.0
 
     if penalty_type == L2:
```

Comment:
In line 539, I think that we can avoid a warning by casting:

```python
        for epoch in range(<Py_ssize_t>max_iter):
```

Reply:
Sure I'll do this.
However just to clarify, the related issue mentions to create a PR to resolve one type of warning for cython module at a time which is why I only looked at -Wcpp for this PR.

Reply:
OK. I see. Since it remains only one. Let's do it in the same PR.